packages/legend-graph/src/graph-manager/action/EngineError.ts

```diff
@@ -16,9 +16,17 @@
 
 import { ApplicationError } from '@finos/legend-shared';
 import type { SourceInformation } from '../action/SourceInformation.js';
+import { makeObservable, observable } from 'mobx';
 
 export class EngineError extends ApplicationError {
   sourceInformation?: SourceInformation | undefined;
+
+  constructor(message: string | undefined) {
+    super(message);
+    makeObservable(this, {
+      sourceInformation: observable,
```

Comment:
we want to avoid adding observable directly on the model. we could add it in its downstream usage?

Reply:
i meant more on the actual creation of it. so maybe we create an observe method and call it here `V1_buildCompilationError`